test/lib/mutation_source_test.cc

```diff
@@ -2710,6 +2710,35 @@ void compare_readers(const schema& s, flat_mutation_reader authority, flat_mutat
     }
 }
 
+static bool compare_readers(const schema& s, flat_mutation_reader_v2& authority, flat_reader_assertions_v2& tested) {
```

Comment:
Are the v1 versions still used after this? If not, you can just refactor them to work on v2 readers instead of adding parallel v2 implementation.

Reply:
Unfortunately both are needed.